src/scan-imports.ts

```diff
@@ -63,34 +64,24 @@ function parseWebModuleSpecifier(specifier: string, knownDependencies: string[])
   if (webModulesIndex === -1) {
     return null;
   }
-  // check if the resolved specifier (including file extension) is a known package.
+
+  // check if the resolved specifier without extension matches npm name format
   const resolvedSpecifier = cleanedSpecifier.substring(webModulesIndex + WEB_MODULES_TOKEN_LENGTH);
-  if (knownDependencies.includes(resolvedSpecifier)) {
-    return resolvedSpecifier;
-  }
-  // check if the resolved specifier (without extension) is a known package.
   const resolvedSpecifierWithoutExtension = stripJsExtension(resolvedSpecifier);
-  if (knownDependencies.includes(resolvedSpecifierWithoutExtension)) {
+  if (validate(resolvedSpecifierWithoutExtension).validForNewPackages) {
```

Comment:
Double-check me here: is this the change we want? After re-reading the problem again, it seems like the original code should have worked.

Unless the module was just being loaded from a different `package.json` than the one Snowpack loaded? In that case, this PR should fix it.

Reply:
Yea, that was the issue: the package.json mentioned the 'package collection', which was just a package containing dependencies for many other packages you might want in a single collection. So you then go to use one of these internally referenced "collection" packages, but then we look for it we can't find it in your package.json (because you only depend on the collection itself, and not the individual package).

Reply:
so just to be clear, the two ways we can go about this are:
1. your logic here: if the specifier without an extension is a valid package name, then it must be a package import, aka there's no way that it could be an import to a file WITHIN a package. Safe to remove the extension and return it.
2. check that the specifier is a package inside of your node_modules directory. If it exists as a folder, it must be a package. Remove extension and return.

One the two, #1 (your logic) is definitely the most performant (no disk read needed). But it's also a bit opaque in terms of why that check guarentees us anything. You should add a comment above explaining basically what I just wrote (feel free to copy/paste if that helps) and why we're doing this check at all.

Does that make sense?